In this project we are going to do the EDA of a dataset containing data about telecome users.

The objective of the project is to identify the factors contributing to customer charm.

In [35]:
#import the required libraries
import pandas as pd

In [37]:
#reading the data from source and creating a dataframe
df = pd.read_csv(r"C:\Users\HP\Desktop\Data-LB\New folder\project2.csv")

In [38]:
#printing top 10 row of dataframe
df.head(10)

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   
5  9305-CDSKC  Female              0      No         No       8          Yes   
6  1452-KIOVK    Male              0      No        Yes      22          Yes   
7  6713-OKOMC  Female              0      No         No      10           No   
8  7892-POOKP  Female              0     Yes         No      28          Yes   
9  6388-TABGU    Male              0      No        Yes      62          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   
5               Yes     Fiber optic             No  ...              Yes   
6               Yes     Fiber optic             No  ...               No   
7  No phone service             DSL            Yes  ...               No   
8               Yes     Fiber optic             No  ...              Yes   
9                No             DSL            Yes  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   
5          No         Yes             Yes  Month-to-month              Yes   
6          No         Yes              No  Month-to-month              Yes   
7          No          No              No  Month-to-month               No   
8         Yes         Yes             Yes  Month-to-month              Yes   
9          No          No              No        One year               No   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  
5           Electronic check          99.65         820.5   Yes  
6    Credit card (automatic)          89.10        1949.4    No  
7               Mailed check          29.75         301.9    No  
8           Electronic check         104.80       3046.05   Yes  
9  Bank transfer (automatic)          56.15       3487.95    No  

[10 rows x 21 columns]

The features and the target can be identify by the visual inspection of above data frame.

* The target variable: The column named 'churn' which contain whether customer is churned out or not(Yes/No)

Apart from the target, they can be further divide into two categories.

* Customer Demography features
* customer services offered 

In [41]:
#print Last 10 row of data frame
df.tail(10)

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
7033  9767-FFLEM    Male              0      No         No      38   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7036  7750-EYXWZ  Female              0      No         No      12   
7037  2569-WGERO  Female              0      No         No      72   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
7033          Yes                No     Fiber optic                   No  ...   
7034          Yes               Yes     Fiber optic                  Yes  ...   
7035          Yes                No     Fiber optic                   No  ...   
7036           No  No phone service             DSL                   No  ...   
7037          Yes                No              No  No internet service  ...   
7038          Yes               Yes             DSL                  Yes  ...   
7039          Yes               Yes     Fiber optic                   No  ...   
7040           No  No phone service             DSL                  Yes  ...   
7041          Yes               Yes     Fiber optic                   No  ...   
7042          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
7033                   No                   No                   No   
7034                  Yes                   No                  Yes   
7035                   No                   No                  Yes   
7036                  Yes                  Yes                  Yes   
7037  No internet service  No internet service  No internet service   
7038                  Yes                  Yes                  Yes   
7039                  Yes                   No                  Yes   
7040                   No                   No                   No   
7041                   No                   No                   No   
7042                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
7033                   No  Month-to-month              Yes   
7034                   No  Month-to-month              Yes   
7035                   No  Month-to-month              Yes   
7036                  Yes        One year               No   
7037  No internet service        Two year              Yes   
7038                  Yes        One year              Yes   
7039                  Yes        One year              Yes   
7040                   No  Month-to-month              Yes   
7041                   No  Month-to-month              Yes   
7042                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
7033    Credit card (automatic)          69.50       2625.25    No  
7034    Credit card (automatic)         102.95       6886.25   Yes  
7035  Bank transfer (automatic)          78.70        1495.1    No  
7036           Electronic check          60.65         743.3    No  
7037  Bank transfer (automatic)          21.15        1419.4    No  
7038               Mailed check          84.80        1990.5    No  
7039    Credit card (automatic)         103.20        7362.9    No  
7040           Electronic check          29.60        346.45    No  
7041               Mailed check          74.40         306.6   Yes  
7042  Bank transfer (automatic)         105.65        6844.5    No  

[10 rows x 21 columns]

In [43]:
#convert all column as lower alphabet
df.columns = [col.lower() for col in df]

In [44]:
#Get the summary of Data Frame, dtype
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerid        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   seniorcitizen     7043 non-null   int64  
 3   partner           7043 non-null   object 
 4   dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   phoneservice      7043 non-null   object 
 7   multiplelines     7043 non-null   object 
 8   internetservice   7043 non-null   object 
 9   onlinesecurity    7043 non-null   object 
 10  onlinebackup      7043 non-null   object 
 11  deviceprotection  7043 non-null   object 
 12  techsupport       7043 non-null   object 
 13  streamingtv       7043 non-null   object 
 14  streamingmovies   7043 non-null   object 
 15  contract          7043 non-null   object 
 16  paperlessbilling  7043 non-null   object 


From the above we can clearly see that our data contains 21 columns and 7043 rows and there is no any missing values in any columns. 

In [45]:
df[df.isnull()].count()

customerid          0
gender              0
seniorcitizen       0
partner             0
dependents          0
tenure              0
phoneservice        0
multiplelines       0
internetservice     0
onlinesecurity      0
onlinebackup        0
deviceprotection    0
techsupport         0
streamingtv         0
streamingmovies     0
contract            0
paperlessbilling    0
paymentmethod       0
monthlycharges      0
totalcharges        0
churn               0
dtype: int64

In [46]:
df.duplicated().sum()

0

In [47]:
df[df.duplicated()].count()

customerid          0
gender              0
seniorcitizen       0
partner             0
dependents          0
tenure              0
phoneservice        0
multiplelines       0
internetservice     0
onlinesecurity      0
onlinebackup        0
deviceprotection    0
techsupport         0
streamingtv         0
streamingmovies     0
contract            0
paperlessbilling    0
paymentmethod       0
monthlycharges      0
totalcharges        0
churn               0
dtype: int64

In [48]:
df.shape

(7043, 21)

In [49]:
df.describe()

seniorcitizen       tenure  monthlycharges
count    7043.000000  7043.000000     7043.000000
mean        0.162147    32.371149       64.761692
std         0.368612    24.559481       30.090047
min         0.000000     0.000000       18.250000
25%         0.000000     9.000000       35.500000
50%         0.000000    29.000000       70.350000
75%         0.000000    55.000000       89.850000
max         1.000000    72.000000      118.750000

In [12]:
df.head(3)

customerid  gender  seniorcitizen partner dependents  tenure phoneservice  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   

      multiplelines internetservice onlinesecurity  ... deviceprotection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   

  techsupport streamingtv streamingmovies        contract paperlessbilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   

      paymentmethod monthlycharges  totalcharges churn  
0  Electronic check          29.85         29.85    No  
1      Mailed check          56.95        1889.5    No  
2      Mailed check          53.85        108.15   Yes  

[3 rows x 21 columns]

In [13]:
df['partner'] =df['partner'].map({'Yes':1,'No':0})

In [14]:
df.head()

customerid  gender  seniorcitizen  partner dependents  tenure phoneservice  \
0  7590-VHVEG  Female              0        1         No       1           No   
1  5575-GNVDE    Male              0        0         No      34          Yes   
2  3668-QPYBK    Male              0        0         No       2          Yes   
3  7795-CFOCW    Male              0        0         No      45           No   
4  9237-HQITU  Female              0        0         No       2          Yes   

      multiplelines internetservice onlinesecurity  ... deviceprotection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  techsupport streamingtv streamingmovies        contract paperlessbilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               paymentmethod monthlycharges  totalcharges churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

In [15]:
df['dependents']=df['dependents'].map({'Yes':1,'No':0})
df['phoneservice']=df['phoneservice'].map({'Yes':1,'No':0})
df['onlinesecurity']=df['onlinesecurity'].map({'Yes':1,'No':0})
df['deviceprotection']=df['deviceprotection'].map({'Yes':1,'No':0})
df['techsupport']=df['techsupport'].map({'Yes':1,'No':0})
df['streamingtv']=df['streamingtv'].map({'Yes':1,'No':0})
df['streamingmovies']=df['streamingmovies'].map({'Yes':1,'No':0})
df['paperlessbilling']=df['paperlessbilling'].map({'Yes':1,'No':0})
df['churn']=df['churn'].map({'Yes':1,'No':0})

In [16]:
df.head(10)

customerid  gender  seniorcitizen  partner  dependents  tenure  \
0  7590-VHVEG  Female              0        1           0       1   
1  5575-GNVDE    Male              0        0           0      34   
2  3668-QPYBK    Male              0        0           0       2   
3  7795-CFOCW    Male              0        0           0      45   
4  9237-HQITU  Female              0        0           0       2   
5  9305-CDSKC  Female              0        0           0       8   
6  1452-KIOVK    Male              0        0           1      22   
7  6713-OKOMC  Female              0        0           0      10   
8  7892-POOKP  Female              0        1           0      28   
9  6388-TABGU    Male              0        0           1      62   

   phoneservice     multiplelines internetservice  onlinesecurity  ...  \
0             0  No phone service             DSL             0.0  ...   
1             1                No             DSL             1.0  ...   
2             1                No             DSL             1.0  ...   
3             0  No phone service             DSL             1.0  ...   
4             1                No     Fiber optic             0.0  ...   
5             1               Yes     Fiber optic             0.0  ...   
6             1               Yes     Fiber optic             0.0  ...   
7             0  No phone service             DSL             1.0  ...   
8             1               Yes     Fiber optic             0.0  ...   
9             1                No             DSL             1.0  ...   

  deviceprotection  techsupport  streamingtv  streamingmovies        contract  \
0              0.0          0.0          0.0              0.0  Month-to-month   
1              1.0          0.0          0.0              0.0        One year   
2              0.0          0.0          0.0              0.0  Month-to-month   
3              1.0          1.0          0.0              0.0        One year   
4              0.0          0.0          0.0              0.0  Month-to-month   
5              1.0          0.0          1.0              1.0  Month-to-month   
6              0.0          0.0          1.0              0.0  Month-to-month   
7              0.0          0.0          0.0              0.0  Month-to-month   
8              1.0          1.0          1.0              1.0  Month-to-month   
9              0.0          0.0          0.0              0.0        One year   

  paperlessbilling              paymentmethod monthlycharges  totalcharges  \
0                1           Electronic check          29.85         29.85   
1                0               Mailed check          56.95        1889.5   
2                1               Mailed check          53.85        108.15   
3                0  Bank transfer (automatic)          42.30       1840.75   
4                1           Electronic check          70.70        151.65   
5                1           Electronic check          99.65         820.5   
6                1    Credit card (automatic)          89.10        1949.4   
7                0               Mailed check          29.75         301.9   
8                1           Electronic check         104.80       3046.05   
9                0  Bank transfer (automatic)          56.15       3487.95   

  churn  
0     0  
1     0  
2     1  
3     0  
4     1  
5     1  
6     0  
7     0  
8     1  
9     0  

[10 rows x 21 columns]

In [17]:
df.iloc[:,8:]

internetservice  onlinesecurity onlinebackup  deviceprotection  \
0                DSL             0.0          Yes               0.0   
1                DSL             1.0           No               1.0   
2                DSL             1.0          Yes               0.0   
3                DSL             1.0           No               1.0   
4        Fiber optic             0.0           No               0.0   
...              ...             ...          ...               ...   
7038             DSL             1.0           No               1.0   
7039     Fiber optic             0.0          Yes               1.0   
7040             DSL             1.0           No               0.0   
7041     Fiber optic             0.0           No               0.0   
7042     Fiber optic             1.0           No               1.0   

      techsupport  streamingtv  streamingmovies        contract  \
0             0.0          0.0              0.0  Month-to-month   
1             0.0          0.0              0.0        One year   
2             0.0          0.0              0.0  Month-to-month   
3             1.0          0.0              0.0        One year   
4             0.0          0.0              0.0  Month-to-month   
...           ...          ...              ...             ...   
7038          1.0          1.0              1.0        One year   
7039          0.0          1.0              1.0        One year   
7040          0.0          0.0              0.0  Month-to-month   
7041          0.0          0.0              0.0  Month-to-month   
7042          1.0          1.0              1.0        Two year   

      paperlessbilling              paymentmethod  monthlycharges  \
0                    1           Electronic check           29.85   
1                    0               Mailed check           56.95   
2                    1               Mailed check           53.85   
3                    0  Bank transfer (automatic)           42.30   
4                    1           Electronic check           70.70   
...                ...                        ...             ...   
7038                 1               Mailed check           84.80   
7039                 1    Credit card (automatic)          103.20   
7040                 1           Electronic check           29.60   
7041                 1               Mailed check           74.40   
7042                 1  Bank transfer (automatic)          105.65   

     totalcharges  churn  
0           29.85      0  
1          1889.5      0  
2          108.15      1  
3         1840.75      0  
4          151.65      1  
...           ...    ...  
7038       1990.5      0  
7039       7362.9      0  
7040       346.45      0  
7041        306.6      1  
7042       6844.5      0  

[7043 rows x 13 columns]

In [18]:
df['onlinebackup']=df['onlinebackup'].map({'Yes':1,'No':0})

In [19]:
df.head()

customerid  gender  seniorcitizen  partner  dependents  tenure  \
0  7590-VHVEG  Female              0        1           0       1   
1  5575-GNVDE    Male              0        0           0      34   
2  3668-QPYBK    Male              0        0           0       2   
3  7795-CFOCW    Male              0        0           0      45   
4  9237-HQITU  Female              0        0           0       2   

   phoneservice     multiplelines internetservice  onlinesecurity  ...  \
0             0  No phone service             DSL             0.0  ...   
1             1                No             DSL             1.0  ...   
2             1                No             DSL             1.0  ...   
3             0  No phone service             DSL             1.0  ...   
4             1                No     Fiber optic             0.0  ...   

   deviceprotection  techsupport  streamingtv  streamingmovies  \
0               0.0          0.0          0.0              0.0   
1               1.0          0.0          0.0              0.0   
2               0.0          0.0          0.0              0.0   
3               1.0          1.0          0.0              0.0   
4               0.0          0.0          0.0              0.0   

         contract paperlessbilling              paymentmethod monthlycharges  \
0  Month-to-month                1           Electronic check          29.85   
1        One year                0               Mailed check          56.95   
2  Month-to-month                1               Mailed check          53.85   
3        One year                0  Bank transfer (automatic)          42.30   
4  Month-to-month                1           Electronic check          70.70   

   totalcharges churn  
0         29.85     0  
1        1889.5     0  
2        108.15     1  
3       1840.75     0  
4        151.65     1  

[5 rows x 21 columns]

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerid        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   seniorcitizen     7043 non-null   int64  
 3   partner           7043 non-null   int64  
 4   dependents        7043 non-null   int64  
 5   tenure            7043 non-null   int64  
 6   phoneservice      7043 non-null   int64  
 7   multiplelines     7043 non-null   object 
 8   internetservice   7043 non-null   object 
 9   onlinesecurity    5517 non-null   float64
 10  onlinebackup      5517 non-null   float64
 11  deviceprotection  5517 non-null   float64
 12  techsupport       5517 non-null   float64
 13  streamingtv       5517 non-null   float64
 14  streamingmovies   5517 non-null   float64
 15  contract          7043 non-null   object 
 16  paperlessbilling  7043 non-null   int64  


In [21]:
df.describe()

seniorcitizen      partner   dependents       tenure  phoneservice  \
count    7043.000000  7043.000000  7043.000000  7043.000000   7043.000000   
mean        0.162147     0.483033     0.299588    32.371149      0.903166   
std         0.368612     0.499748     0.458110    24.559481      0.295752   
min         0.000000     0.000000     0.000000     0.000000      0.000000   
25%         0.000000     0.000000     0.000000     9.000000      1.000000   
50%         0.000000     0.000000     0.000000    29.000000      1.000000   
75%         0.000000     1.000000     1.000000    55.000000      1.000000   
max         1.000000     1.000000     1.000000    72.000000      1.000000   

       onlinesecurity  onlinebackup  deviceprotection  techsupport  \
count     5517.000000   5517.000000       5517.000000  5517.000000   
mean         0.365960      0.440276          0.439007     0.370491   
std          0.481742      0.496465          0.496311     0.482980   
min          0.000000      0.000000          0.000000     0.000000   
25%          0.000000      0.000000          0.000000     0.000000   
50%          0.000000      0.000000          0.000000     0.000000   
75%          1.000000      1.000000          1.000000     1.000000   
max          1.000000      1.000000          1.000000     1.000000   

       streamingtv  streamingmovies  paperlessbilling  monthlycharges  \
count  5517.000000      5517.000000       7043.000000     7043.000000   
mean      0.490665         0.495197          0.592219       64.761692   
std       0.499958         0.500022          0.491457       30.090047   
min       0.000000         0.000000          0.000000       18.250000   
25%       0.000000         0.000000          0.000000       35.500000   
50%       0.000000         0.000000          1.000000       70.350000   
75%       1.000000         1.000000          1.000000       89.850000   
max       1.000000         1.000000          1.000000      118.750000   

             churn  
count  7043.000000  
mean      0.265370  
std       0.441561  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       1.000000  
max       1.000000

In [22]:
df.corr()

seniorcitizen   partner  dependents    tenure  phoneservice  \
seniorcitizen          1.000000  0.016479   -0.211185  0.016567      0.008576   
partner                0.016479  1.000000    0.452676  0.379697      0.017706   
dependents            -0.211185  0.452676    1.000000  0.159712     -0.001762   
tenure                 0.016567  0.379697    0.159712  1.000000      0.008448   
phoneservice           0.008576  0.017706   -0.001762  0.008448      1.000000   
onlinesecurity        -0.110473  0.171751    0.158531  0.376008     -0.038202   
onlinebackup          -0.003577  0.173251    0.097615  0.421591      0.014718   
deviceprotection      -0.011482  0.188201    0.085130  0.421932     -0.006211   
techsupport           -0.135584  0.144218    0.137187  0.372711     -0.041421   
streamingtv            0.033852  0.155179    0.053502  0.326901      0.054675   
streamingmovies        0.050292  0.146401    0.024245  0.335160      0.043754   
paperlessbilling       0.156530 -0.014877   -0.111377  0.006152      0.016505   
monthlycharges         0.220173  0.096848   -0.113890  0.247900      0.247398   
churn                  0.150889 -0.150448   -0.164221 -0.352229      0.011942   

                  onlinesecurity  onlinebackup  deviceprotection  techsupport  \
seniorcitizen          -0.110473     -0.003577         -0.011482    -0.135584   
partner                 0.171751      0.173251          0.188201     0.144218   
dependents              0.158531      0.097615          0.085130     0.137187   
tenure                  0.376008      0.421591          0.421932     0.372711   
phoneservice           -0.038202      0.014718         -0.006211    -0.041421   
onlinesecurity          1.000000      0.179711          0.170336     0.273472   
onlinebackup            0.179711      1.000000          0.185155     0.190586   
deviceprotection        0.170336      0.185155          1.000000     0.235716   
techsupport             0.273472      0.190586          0.235716     1.000000   
streamingtv             0.043918      0.146935          0.276617     0.161478   
streamingmovies         0.055842      0.136687          0.289019     0.162011   
paperlessbilling       -0.139168      0.005437         -0.023632    -0.088224   
monthlycharges          0.069118      0.252337          0.321927     0.134233   
churn                  -0.280816     -0.196061         -0.177125    -0.274422   

                  streamingtv  streamingmovies  paperlessbilling  \
seniorcitizen        0.033852         0.050292          0.156530   
partner              0.155179         0.146401         -0.014877   
dependents           0.053502         0.024245         -0.111377   
tenure               0.326901         0.335160          0.006152   
phoneservice         0.054675         0.043754          0.016505   
onlinesecurity       0.043918         0.055842         -0.139168   
onlinebackup         0.146935         0.136687          0.005437   
deviceprotection     0.276617         0.289019         -0.023632   
techsupport          0.161478         0.162011         -0.088224   
streamingtv          1.000000         0.434754          0.117895   
streamingmovies      0.434754         1.000000          0.100942   
paperlessbilling     0.117895         0.100942          1.000000   
monthlycharges       0.532439         0.525495          0.352150   
churn               -0.037057        -0.040132          0.191825   

                  monthlycharges     churn  
seniorcitizen           0.220173  0.150889  
partner                 0.096848 -0.150448  
dependents             -0.113890 -0.164221  
tenure                  0.247900 -0.352229  
phoneservice            0.247398  0.011942  
onlinesecurity          0.069118 -0.280816  
onlinebackup            0.252337 -0.196061  
deviceprotection        0.321927 -0.177125  
techsupport             0.134233 -0.274422  
streamingtv             0.532439 -0.037057  
streamingmovies         0.525495 -0.040132  
paperlessbilling        0

In [23]:
df['onlinebackup'].corr(df['churn'])

-0.196061208316123

In [24]:
df['partner'].corr(df['churn'])

-0.15044754495917612

In [25]:
df['dependents'].corr(df['churn'])

-0.1642214015797272

In [26]:
df['tenure'].corr(df['churn'])

-0.3522286701130774

In [27]:
df['phoneservice'].corr(df['churn'])

0.011941980029002954

In [28]:
df['onlinesecurity'].corr(df['churn'])

-0.28081637309329366

In [29]:
df['deviceprotection'].corr(df['churn'])

-0.17712496477172568

In [30]:
df['techsupport'].corr(df['churn'])

-0.2744218156579737

In [31]:
df['techsupport'].corr(df['churn'])

-0.2744218156579737

In [32]:
df['paperlessbilling'].corr(df['churn'])

0.19182533166646804

In [33]:
df['monthlycharges'].corr(df['churn'])

0.19335642223784671